DNN Platform/DotNetNuke.DependencyInjection/IServiceRegistration.cs

```diff
@@ -0,0 +1,9 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+
+namespace DotNetNuke.DependencyInjection
+{
+    public interface IServiceRegistration
```

Comment:
Can we get XML documentation for this so users will have intelligence detail.

Reply:
I am going to rename `IServiceRegistration` -> `IDnnStartup` to be more consistent with the `Microsoft.AspNetCore.Hosting.IStartup` API (which isn't included yet).  In AspNetCore you can create a `Startup` file by convention or use the `IStartup` interface. The `IDnnStartup` interface will have the convention method of `ConfigureServices` which will be invoked correctly as-is.

Using the `IStartup` interface really isn't an option right now and isn't a good idea as that interface returns an `IServicePrincipal` which won't work well with how many instances of `IDnnStartup` there are going to be when we really take advantage of Dependency Injection.

Here is the AspNetCore Startup logic for reference. 
```c#
public static IWebHostBuilder UseStartup(this IWebHostBuilder hostBuilder, Type startupType)
{
    var startupAssemblyName = startupType.GetTypeInfo().Assembly.GetName().Name;

    hostBuilder.UseSetting(WebHostDefaults.ApplicationKey, startupAssemblyName);

    // Light up the GenericWebHostBuilder implementation
    if (hostBuilder is ISupportsStartup supportsStartup)
    {
        return supportsStartup.UseStartup(startupType);
    }

    return hostBuilder
        .ConfigureServices(services =>
        {
            if (typeof(IStartup).GetTypeInfo().IsAssignableFrom(startupType.GetTypeInfo()))
            {
                services.AddSingleton(typeof(IStartup), startupType);
            }
            else
            {
                services.AddSingleton(typeof(IStartup), sp =>
                {
                    var hostingEnvironment = sp.GetRequiredService<IHostEnvironment>();
                    return new ConventionBasedStartup(StartupLoader.LoadMethods(sp, startupType, hostingEnvironment.EnvironmentName));
                });
            }
        });
}
```

The most common usage in AspNetCore is not using the `IStartup` interface, and I really think in an AspNetCore version of Dnn we should continue to use our own `IDnnStartup` to give us access to our customized startup code for both module pipelines and library code.

Reply:
Added xml comments